### PR TITLE
VMware: move vmware module_utils

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
vmware.py from module_utils is moved to vmware_common directory
as vmware namespace is colliding with other VMware official
libraries e.g. vSphere Automation SDK.
vmware.py has new home under vmware_common.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lots of them

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@abadger @bcoca @tchernomax @pdellaert @dagwieers @jctanner Could you please review this ? This change is required as I am introducing new vmware module_utils for REST calls which is based upon vSphere Automation SDK. There is clash between vmware.py and vSphere Automation SDK namespaces, so moving vmware.py in vmware_common directory. Please refer this [vmware_tag](https://github.com/ansible/ansible/pull/37261) PR for more details.

**Comment by Akasurde** [comment]
Can someone review this ?

**Comment by gundalow** [comment]
You may wish to add this new directory into BOTMETA.yml as well

**Comment by gundalow** [comment]
BOTMETA.yml for maintainers & to add vmware label

**Review by pdellaert — APPROVED** [review]
LGTM

File: lib/ansible/module_utils/vmware_common/vmware.py
Comment (webknjaz) [inline lib/ansible/module_utils/vmware_common/vmware.py]: How about doing a [truncated exponential backoff](https://cloud.google.com/storage/docs/exponential-backoff) here?

Comment (Akasurde) [reply lib/ansible/module_utils/vmware_common/vmware.py]: @webknjaz This is a legacy code and I will address this in different PR.

**Comment by Akasurde** [comment]
@gundalow Done.

**Comment by abadger** [comment]
How long has module_utils/VMware.py existed?  Maybe you need to include a
VMware.py at the old location for backwards compatiblity.  (Import all the
public symbols from the new location.  It would be good to return a
deprecaution warning as well but you won't have access to an ansible module
object when it's imported so I don't know that there's a way to do that....

**Comment by abadger** [comment]
Why not module_utils/vmware/common.py ?

**Comment by mattclay** [comment]
@Akasurde Can you provide an example of what the conflict is?

**Comment by Akasurde** [comment]
As per @abadger's suggestion, I added boilerplate at top of [module_utils/vmware.py](https://github.com/ansible/ansible/pull/37261/files#diff-d5ca044435f1fc628995b038d2e238f7R20) and [module_utils/vmware_rest_client.py](https://github.com/ansible/ansible/pull/37261/files#diff-fabcac47d095dbb55d73f01d86a91d1bR6) which resolved the issue of ambiguous import in both vmware.py and vmware_rest_client.py. 
I will close this PR once #37261 is merged.

**Comment by Akasurde** [comment]
Closing this PR as import boilerplate fixes the import conflicts. Thanks everyone for reviews and comments.

**Pull request closed without merging**
